### PR TITLE
[SPARK-40049][SQL][TESTS] Add ReplaceNullWithFalseInPredicateWithAQEEndToEndSuite

### PR DESCRIPTION
**Pull request by kazuyukitanimura**
### What changes were proposed in this pull request?
This PR proposes to add `ReplaceNullWithFalseInPredicateWithAQEEndToEndSuite ` i.e. test cases of `ReplaceNullWithFalseInPredicateEndToEndSuite` with AQE (Adaptive Query Execution) turned on.

### Why are the changes needed?
Currently `ReplaceNullWithFalseInPredicateEndToEndSuite` assumes that AQE is always turned off. We should also test with AQE turned on

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Added an AQE version tests along with the non AQE version

### CODE REVIEW

**Comment by kazuyukitanimura** [comment]
cc @aokolnychyi @wangyum @sunchao

**Review by sunchao — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala [hunk 1]
```diff
@@ -19,16 +19,25 @@ package org.apache.spark.sql
 
 import org.apache.spark.sql.catalyst.expressions.{CaseWhen, If, Literal}
 import org.apache.spark.sql.execution.LocalTableScanExec
+import org.apache.spark.sql.execution.adaptive.{
+  AdaptiveSparkPlanHelper,
+  DisableAdaptiveExecutionSuite,
+  EnableAdaptiveExecutionSuite
+}
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala:26]: Hi, @kazuyukitanimura . This is not Apache Spark style. Please match with the code around your patch.

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala:26]: I'm wondering if this came from IDE setting.

File: sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala [hunk 2]
```diff
@@ -19,16 +19,25 @@ package org.apache.spark.sql
...
 import org.apache.spark.sql.functions.{lit, when}
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.test.SharedSparkSession
 import org.apache.spark.sql.types.BooleanType
 
-class ReplaceNullWithFalseInPredicateEndToEndSuite extends QueryTest with SharedSparkSession {
+class ReplaceNullWithFalseInPredicateWithAQEEndToEndSuite extends
+  ReplaceNullWithFalseInPredicateEndToEndSuite with EnableAdaptiveExecutionSuite
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala:33]: Please move this class to the end of this file.

File: sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala (hunk 1 above)
Comment (kazuyukitanimura) [reply sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala]: Updated

File: sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala (hunk 2 above)
Comment (kazuyukitanimura) [reply sql/core/src/test/scala/org/apache/spark/sql/ReplaceNullWithFalseInPredicateEndToEndSuite.scala]: Updated

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @kazuyukitanimura , @sunchao .

**Comment by cloud-fan** [comment]
Hey, are we going to update every test suite to run with both AQE on and off? I don't think it's worthwhile since AQE is already on by default.

**Comment by kazuyukitanimura** [comment]
Yes AQE is on by default, however `spark.sql.adaptive.forceApply` is off by default. Enabling force apply changes the execution plan. Perhaps I should have been clear for this point.

**Pull request closed without merging**